### PR TITLE
HuggingFace support for Quality Classifier

### PR DESCRIPTION
**Pull request by sarahyurick**
Closes https://github.com/NVIDIA/NeMo-Curator/issues/167

### CODE REVIEW

File: nemo_curator/classifiers/quality.py [hunk 1]
```diff
@@ -12,22 +12,21 @@
...
+    HFCustomModel,
     _run_classifier_helper,
 )
 from nemo_curator.datasets import DocumentDataset
 
+QUALITY_IDENTIFIER = "nvidia/quality-classifier-deberta"
+
 
 @dataclass
 class QualityModelConfig:
```
Comment (sarahyurick) [inline nemo_curator/classifiers/quality.py:32]: There's a lot of repeated/identical logic for `DomainModelConfig` vs `QualityModelConfig`, `DomainModel` vs `QualityModel`, and `DomainClassifier` vs `QualityClassifier`. I was wondering how much of this I should try to consolidate versus keeping them separate?

**Comment by sarahyurick** [comment]
Hi @VibhuJawa @ryantwolf this is ready for review.

File: nemo_curator/classifiers/base.py [hunk 2]
```diff
@@ -78,6 +83,32 @@ def get_labels(self) -> List[str]:
         return self.labels
 
 
+class HFCustomModel(nn.Module, PyTorchModelHubMixin):
```
Comment (ryantwolf) [inline nemo_curator/classifiers/base.py:86]: Can you rename this to be something like `HFDeberta`?

File: nemo_curator/classifiers/quality.py [hunk 3]
```diff
@@ -36,108 +35,28 @@ class QualityModelConfig:
...
-        if self.autocast:
-            with torch.autocast(device_type="cuda"):
-                return self._forward(batch)
-        else:
-            return self._forward(batch)
-
-
 class QualityModel(HFModel):
-    def __init__(self, config, out_dim=None, model_path=None, autocast=False):
+    def __init__(self, config: dataclass, autocast: bool = False):
```
Comment (ryantwolf) [inline nemo_curator/classifiers/quality.py:39]: Shoot I meant to do this in my PR, but can you change the type annotation here from `dataclass` to `QualityModelConfig`? Same with the domain classifier.

**Review by ryantwolf — COMMENTED** [review]
Thanks for the quick turnaround on this! I just have two small nits.

File: nemo_curator/classifiers/quality.py (hunk 1 above)
Comment (ryantwolf) [reply nemo_curator/classifiers/quality.py:32]: I think you've done a good job so far. We'll keep it with what you have for now.

File: nemo_curator/classifiers/quality.py [hunk 4]
```diff
@@ -36,108 +35,28 @@ class QualityModelConfig:
     max_len = 512
```
Comment (ryantwolf) [inline nemo_curator/classifiers/quality.py:35]: This should be updated too. This model has `max_len = 1024`

**Comment by sarahyurick** [comment]
@ryantwolf updated, thanks!

**Review by ryantwolf — APPROVED** [review]
Thanks Sarah!

File: nemo_curator/classifiers/domain.py [hunk 5]
```diff
@@ -37,40 +35,14 @@ class DomainModelConfig:
     max_len = 512
```
Comment (VibhuJawa) [inline nemo_curator/classifiers/domain.py:35]: Minor non blocking NIT

@ryantwolf / @sarahyurick , Are we sure the `max_len` for domain is `512` and `quality` is `1024`. Might be worth double checking this.

**Review by VibhuJawa — APPROVED** [review]
LGTM, thanks for working on this Sarah. 

Have added a non blocking nit, which i dont think is related to your PR

File: nemo_curator/classifiers/domain.py (hunk 5 above)
Comment (ryantwolf) [reply nemo_curator/classifiers/domain.py:35]: Double checked, this is correct.

**Pull request merged**